tests/Unit/ParallelAlgorithms/Interpolation/Test_InterpolatorReceiveAndDumpVolumeData.cpp

```diff
@@ -313,12 +317,34 @@ struct MockMetavariables {
         intrp::callbacks::ObserveTimeSeriesOnSurface<tmpl::list<>,
                                                      InterpolationTargetA>;
   };
+  struct InterpolationTargetB
+      : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
+    using temporal_id = ::Tags::TimeStepId;
+    using vars_to_interpolate_to_target = tmpl::list<Tags::Square>;
+    using compute_vars_to_interpolate = ComputeSquare;
+    using compute_items_on_target = tmpl::list<>;
+    using compute_target_points =
+        ::intrp::TargetPoints::LineSegment<InterpolationTargetB, 3>;
+    using post_interpolation_callback =
+        intrp::callbacks::ObserveTimeSeriesOnSurface<tmpl::list<>,
+                                                     InterpolationTargetB>;
+    // This just needs to be something other than the interpolator. Doesn't
+    // matter what because we won't actually be doing an interpolation to this
+    // component. We just needed another target with the same temporal_id that
+    // wasn't using the interpolator to test that InterpolatorReceiveVolumeData
+    // doesn't hold onto volume data for this temporal_id that will never be
+    // used by this target.
+    template <typename Metavariables>
+    using interpolating_component = mock_observer_writer<Metavariables>;
+  };
```

Comment:
Maybe I'm being pedantic, but what if instead you avoided duplicating code by using a template parameter and `if constexpr` to decide whether or not to add the `interpolating_component` ... otherwise, aren't the two targets the same?